staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -569,6 +617,54 @@ func (s *sharedIndexInformer) HandleDeltas(obj interface{}) error {
 	return nil
 }
 
+// IsStarted reports whether the informer has already been started
+func (s *sharedIndexInformer) IsStarted() bool {
+	s.startedLock.Lock()
+	defer s.startedLock.Unlock()
+	return s.started
+}
+
+// IsStopped reports whether the informer has already been stopped
+func (s *sharedIndexInformer) IsStopped() bool {
+	s.startedLock.Lock()
+	defer s.startedLock.Unlock()
+	return s.stopped
+}
+
+// EventHandlerCount reports whether the informer still has registered
+// event handlers
+func (s *sharedIndexInformer) EventHandlerCount() int {
+	s.startedLock.Lock()
+	defer s.startedLock.Unlock()
+	return len(s.processor.listeners)
+}
+
+// RemoveEventHandlerByHandle tries to remove a formerly added event handler by its
+// handle returned for its registration.
+// If a handler has been added multiple times, only the registration for the
+// given handle will be removed.
+func (s *sharedIndexInformer) RemoveEventHandlerByHandle(handle *ResourceEventHandlerHandle) error {
+	s.startedLock.Lock()
+	defer s.startedLock.Unlock()
+
+	if handle.listener == nil {
+		return fmt.Errorf("registration already removed")
+	}
```

Comment:
move this check before the lock. I also think this should probably return nil, since the desired state (the listener associated with the handle is removed) still holds if there is no associated listener (it makes calling remove twice on the same handle an idempotent operation)

Reply:
I've additionally added a method to the handle to check the handle state